builtin/logical/pki/path_roles.go

```diff
@@ -1116,3 +1117,42 @@ const pathListRolesHelpDesc = `Roles will be listed by the role name.`
 const pathRoleHelpSyn = `Manage the roles that can be created with this backend.`
 
 const pathRoleHelpDesc = `This path lets you manage the roles that can be created with this backend.`
+
+const policyIdentifiersParam = "policy_identifiers"
+
+func getPolicyIdentifier(data *framework.FieldData, defaultIdentifiers *[]string) []string {
+	policyIdentifierEntry, ok := data.GetOk(policyIdentifiersParam)
+	if !ok {
+		// No Entry for policy_identifiers
+		if defaultIdentifiers != nil {
+			return *defaultIdentifiers
+		}
+		return data.Get(policyIdentifiersParam).([]string)
+	}
+	// Could Be A JSON Entry
+	policyIdentifierJsonEntry := data.Raw[policyIdentifiersParam]
+	policyIdentifiers, err := parsePolicyIdentifiersFromJson(policyIdentifierJsonEntry.(string))
+	if err == nil {
+		return policyIdentifiers
+	}
+	// Else could Just Be A List of OIDs
+	return policyIdentifierEntry.([]string)
+}
+
+func parsePolicyIdentifiersFromJson(policyIdentifiers string) ([]string, error) {
+	var entries []certutil.PolicyIdentifierWithQualifierEntry
+	var policyIdentifierList []string
+	err := json.Unmarshal([]byte(policyIdentifiers), &entries)
+	if err != nil {
+		return policyIdentifierList, err
+	}
+	policyIdentifierList = make([]string, len(entries))
```

Comment:
I *believe* you want the following form of make? As I think the append below will start at len(entries) and not 0?

```suggestion
	policyIdentifierList = make([]string, 0, len(entries))
```

Reply:
Thank you!  That solves where my nil policyIdentifiers were coming from :)